### ISSUE TITLE
Extension installation via symlink not working

### ISSUE THREAD

**Issue by romibi** [FreshRSS/FreshRSS#6040]
I finally switching from my own modified Freshrss with "hardcoded" QuickCollapse" button to the normal FreshRSS with the QuickCollapse Extension.
In order to be able to easily update, I checked out this Extension repo next to the FreshRSS folder and added the Extension as a symlink:

```
www-data@***:~/FreshRSS/extensions$ ls -la
total 16
drwxrwxr-x  2 www-data www-data 4096 Jan 11 12:13 .
drwxr-xr-x 14 www-data www-data 4096 Jan 11 12:00 ..
-rw-rw-r--  1 www-data www-data    6 Nov  9 16:19 .gitignore
-rw-rw-r--  1 www-data www-data  197 Nov  9 16:19 README.md
lrwxrwxrwx  1 www-data www-data   50 Jan 11 12:13 xExtension-QuickCollapse -> ../../FreshRSSExtensions/xExtension-QuickCollapse/
```

This seemd to work fine at first. The extension appeared in the config.
But it didn't work.

When analyzing the issue with the Browser Dev Tools i noticed that the javascript wasn't loaded.
When opening `<domain>/freshrss/ext.php?f=xExtension-QuickCollapse%2Fstatic%2Fscript.js&t=js&1704970022` instead of the javascript file i only get:
File is not supported

I have now removed the symlink and just copied the folder over to the extensions folder …
Now it works fine.

It would be nice if this would be possible if it isn't too difficult to fix.

Btw: I think it would make sense to ignore anything in the extensions folder, not only stuff startig with "x". But I don't know if it is worth it's own issue in the FreshRSS repo.

Edit: This was with FreshRSS 1.22.1. After installing the extension I have now realized there was a new release.
I have now updated to 1.23.1 but not tested this issue again.

**Comment by Alkarex** [comment]
Hello,
What installation do you have? Docker? If not, what Web server and OS?

**Comment by romibi** [comment]
No Docker:
Here's some version outputs:
```
> lshw -short | grep system
                        system         Raspberry Pi 4 Model B Rev 1.1
> lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy
> apache2 -v
Server version: Apache/2.4.52 (Ubuntu)
Server built:   2023-10-26T13:44:44
> php -version
PHP 8.1.2-1ubuntu2.14 (cli) (built: Aug 18 2023 11:41:11) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.2-1ubuntu2.14, Copyright (c), by Zend Technologies
```
And the relevant snippet from the apache conf:
```
        Alias /freshrss "/var/www/FreshRSS/p/"
        <Directory "/var/www/FreshRSS/p/">
                Options Indexes FollowSymlinks MultiViews
                AllowOverride None
                Require all granted
        </Directory>
```
On the install check page everything is green.

**Comment by Alkarex** [comment]
Have you given Apache access to `/var/www/FreshRSSExtensions/` ?

1. Check the rights with `ls -alh /var/www/FreshRSSExtensions/`
2. Check your Apache config for `<Directory /var/www/FreshRSSExtensions/>` (or higher up)

**Comment by romibi** [comment]
Yes Apache has access. I've switched back to the symlink now and temporarily added the following block to the apache config:
```
        Alias /freshrssExt "/var/www/FreshRSSExtensions/"
        <Directory "/var/www/FreshRSSExtensions/">
                Options Indexes FollowSymlinks MultiViews
                AllowOverride None
                Require all granted
        </Directory>
        Alias /freshrssExtLn "/var/www/FreshRSS/extensions/"
        <Directory "/var/www/FreshRSSExtensions/">
                Options Indexes FollowSymlinks MultiViews
                AllowOverride None
                Require all granted
        </Directory>
```
Both `<domain>/freshrssExt/xExtension-QuickCollapse/static/script.js` and `<domain>/freshrssExtLn/xExtension-QuickCollapse/static/script.js` are accessible. 
`<domain>/freshrss/ext.php?f=xExtension-QuickCollapse%2Fstatic%2Fscript.js&t=js&1704970022` returns "File is not supported."
with no html just this text even though `Content-Type: text/html; charset=UTF-8`.


If I check out the code in ext.php near that error message:
```
$absolute_filename = get_absolute_filename($file_name);
if (!is_valid_path($absolute_filename)) {
        sendBadRequestResponse('File is not supported.');
}
```
`get_absolute_filename($file_name)` seems to return `/var/www/FreshRSSExtensions/xExtension-QuickCollapse/static/script.js`.
`is_valid_path($absolute_filename)` checks that its in allowed folders, which makes sense for security concerns …
Now either an absolute but non-resolved path should be checked or there should be a way to add folders to the allowed ones …
